WebApiClient/controllers/MultiImageGeneratorController.cs

```diff
@@ -70,4 +82,82 @@ public async Task<ObjectResult> QueryImage(ImageQueryRequest imageQueryRequest)
         return StatusCode(200, new ImageQueryResponseOk { Images = images });
 
     }
+    [HttpPost("process")]
+    public IActionResult AddWatermark([FromBody] WatermarkApiSchema.WatermarkRequest request)
```

Comment:
You should create a grain for this. Application logic should be in grains. The controller only calls the grain to process.